src/jit/lsraxarch.cpp

```diff
@@ -821,6 +821,9 @@ bool LinearScan::isRMWRegOper(GenTree* tree)
         case GT_MUL:
             return (!tree->gtOp.gtOp2->isContainedIntOrIImmed() && !tree->gtOp.gtOp1->isContainedIntOrIImmed());
 
+        case GT_HWIntrinsic:
+            return ((Compiler::flagsOfHWIntrinsic(tree->AsHWIntrinsic()->gtHWIntrinsicId) & HW_Flag_NoRMWSemantics) == 0);
```

Comment:
Doesn't `HW_Flag_NoRMWSemantics` only impact SSE encoding?

Reply:
Shall we remove `HW_Flag_NoRMWSemantics` from all the AVX/AVX2 intrinsics that have VEX.NDS encoding?

Reply:
Good catch. Will fix.

Reply:
> Shall we remove HW_Flag_NoRMWSemantics from all the AVX/AVX2 intrinsics that have VEX.NDS encoding?

We probably want to, that will also require updating the `BuildHWIntrinsic` to account for this.

Reply:
Could you clarify? I'm probably confused, but it would seem that if you remove it from all the AVX intrinsics, then they will be treated as RMW. You don't want to have to check both the Isa and this flag. Maybe this is a good time to consider changing the sense of the flag.

Reply:
@CarolEidt, currently the flag is set on a number of intrinsics but is ignored if the compiler supports the VEX-encoding.

This behavior is generally correct, but misses a couple of instructions which, even under VEX, still have RMW semantics because they don't have the VEX.NDS flag (which specifies the source registers will be preserved).

Instructions such as `CVTDQ2PD` fall into this category.

Reply:
> currently the flag is set on a number of intrinsics but is ignored if the compiler supports the VEX-encoding.

Right, but it means that you have to check both the supported ISA and the flag. Though I guess that, since we'll use the VEX-encoding even for SSE* operations, both checks are required in order to generate the best code.

Reply:
Right. Currently we have to check both `compiler->canUseVexEncoding` and the flag, since the actual behavior depends on both.